client/engine/chainservice/eth_chainservice.go

```diff
@@ -123,8 +124,17 @@ func (ecs *EthChainService) listenForLogEvents(na *NitroAdjudicator.NitroAdjudic
 					panic(err)
 				}
 
-				// TODO set Holdings based on chain event
-				event := AllocationUpdatedEvent{CommonEvent: CommonEvent{channelID: au.ChannelId, BlockNum: chainEvent.BlockNumber}, Holdings: types.Funds{}}
+				tx, pending, err := chain.TransactionByHash(context.Background(), chainEvent.TxHash)
+				if pending || err != nil {
+					panic("Expected transacion to be part of the chain")
```

Comment:
I think we want to avoid panicking here. I suppose the condition could be triggered by a re-org? In any case, we don't want to crash the whole wallet. It would be better to log the problem and carry on (for now, until we come up with a better error handling strategy).

Reply:
You are pointing out a larger problem here. We don't have a strategy for chain service error handling. Perhaps we should address this in a future PR instead of starting to mix logs vs panics for errors?